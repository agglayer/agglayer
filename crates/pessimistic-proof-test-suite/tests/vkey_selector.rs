#[test]
fn vkey_snapshot() {
    let selector = hex::encode(pessimistic_proof::core::PESSIMISTIC_PROOF_PROGRAM_SELECTOR);
    let vkey = agglayer_prover::compute_program_vkey(pessimistic_proof::ELF);

    let message: String = [
        "# If this test fails, it means the PP vkey has changed.",
        "# When that happens, consider updating the selector by bumping the PP version.",
        &format!("| PP_VKEY          | {vkey} |"),
        &format!("| PP_VKEY_SELECTOR | 0x{selector:64} |"),
    ]
    .into_iter()
    .flat_map(|line| [line, "\n"])
    .collect();

    insta::assert_snapshot!(message);
}
