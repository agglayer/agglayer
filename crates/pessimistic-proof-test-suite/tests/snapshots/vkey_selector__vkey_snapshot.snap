---
source: crates/pessimistic-proof-test-suite/tests/vkey_selector.rs
expression: message
---
# If this test fails, it means the PP vkey has changed.
# When that happens, consider updating the selector by bumping the PP version.
| PP_VKEY          | 0x00810a4bc7bb144509a1694ff4c22d37025b9850f0bf58ed800845616934e51c |
| PP_VKEY_SELECTOR | 0x00000006                                                         |
